src/rngs/thread.rs

```diff
@@ -92,23 +95,30 @@ impl Default for ThreadRng {
     }
 }
 
+impl ThreadRng {
+    #[inline(always)]
+    fn rng(&mut self) -> &mut ReseedingRng<Core, OsRng> {
+        unsafe { &mut *self.rng.get() }
```

Comment:
I think there's a subtle safety argument here: the caller is required to stop using the return mutable reference before anyone else calls this method. That is crucial to avoid aliasing mutable references.

That might be worth documenting?

Reply:
You're right of course. It's not a public function so I didn't put much thought into its signature.

Reply:
I submitted #1037 to fix this.

Reply:
@RalfJung Which function should be `unsafe`?

Reply:
(I answered there.)